src/hdrgen.d

```diff
@@ -1742,10 +1742,11 @@ public:
         //printf("FuncDeclaration::toCBuffer() '%s'\n", f.toChars());
         if (stcToBuffer(buf, f.storage_class))
             buf.writeByte(' ');
-        typeToBuffer(f.type, f.ident);
+        auto tf = cast(TypeFunction)f.type;
+        typeToBuffer(tf, f.ident);
         if (hgs.hdrgen == 1)
         {
-            if (f.storage_class & STCauto)
+            if (!f.type.nextOf || f.storage_class & STCauto)
```

Comment:
Hmm ....
since you have the tf variable already,
tf.next will do fine and it a tiny bit more efficient.